draft rel notes for edge 93 (#1441)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/release-notes.md

> To use a prerelease SDK along with a Microsoft Edge preview channel, navigate to [Switch to a preview channel to test upcoming APIs and features](how-to/set-preview-channel.md).

Reviewer: update date at top of file

Author: done

<!-- ## 1.0.1010-prerelease

Release Date: September 13, 2021  

[NuGet package for WebView2 SDK 1.0.955-prerelease][NuGetGallery1.0.955-prerelease]

For full API compatibility, this prerelease version of the WebView2 SDK requires Microsoft Edge version 94.0.1010.0 or higher.

### General  

> *   General WebView2 performance improvements.

Reviewer 2: ```suggestion
*   WebView2 performance improvements.
```

> *   General reliability fixes. \([\#1605][GithubMicrosoftedgeWebviewfeedbackIssue1605] and[\#1678][GithubMicrosoftedgeWebviewfeedbackIssue1678]\)

Reviewer 2: ```suggestion
*   Reliability fixes. \([\#1605][GithubMicrosoftedgeWebviewfeedbackIssue1605] and[\#1678][GithubMicrosoftedgeWebviewfeedbackIssue1678]\)
```

> *   Added launch and foreground boost support.

Reviewer 2: Can we clarify this statement?

#### Experimental Features  

> *   `EnsureCoreWebView2Async` will now throw an `ArgumentException` if called multiple times with incompatible parameters, such as new environments, to avoid silent failures.

Reviewer 2: ```suggestion
*   Removal of silent failures by using `EnsureCoreWebView2Async`, which throws an `ArgumentException` when called multiple times with incompatible parameters.
```

> *   Added [UserDataFolder][Webview2ReferenceWin32Icorewebview2experimentalenvironment5ViewWebview210999PrereleaseGetUserdatafolder] property to Environment Interface. 

Reviewer 3: We're changing the default handling for the user data folder. Maybe we should add something more descriptive and link to this announcement: https://github.com/MicrosoftEdge/WebView2Announcements/issues/26

> *   Added [UserDataFolder][Webview2ReferenceWin32Icorewebview2experimentalenvironment5ViewWebview210999PrereleaseGetUserdatafolder] property to Environment Interface. 

Reviewer 2: Can we link to what environment interface is?

*   Added [navigation & script APIs][Webview2ReferenceWin32Icorewebview2experimentalframeViewWebview210999Prerelease] for iFrames. 

> *   Added MemoryUsageTargetLevel API which allows developer to specify desired memory consumption level (low, normal). 

Reviewer 2: ```suggestion
*   Added MemoryUsageTargetLevel which allows developers to specify memory consumption levels, such as low, or normal. 
```

> *   Added ExclusiveUserDataFolderAccess to EnvironmentOptions.

Reviewer 3: Is this part of the UDF handling breaking change listed above? We should probably combine both if it is.

Author: not related to UDF breaking changes.

> *   Added HiddenPdfToolbarItems API to customize pdf toolbar items.

Reviewer 2: ```suggestion
*   Added HiddenPdfToolbarItems to customize pdf toolbar items.
```

Reviewer 2: What does "pdf toolbar items" refer to?

> *   Added PrintToPDF API which allows printing the current page to PDF with optional custom settings.

Reviewer 2: ```suggestion
*   Added PrintToPDF, which allows printing the current page to PDF. Also, you can use optional custom settings with this new API.
```

> *   Added AllowExternalDrop property to enable/disable drag and drop of objects outside of webview2 into it.

Reviewer 2: ```suggestion
*   Added `AllowExternalDrop` property to allow the dragging and dropping of objects from outside a WebView2 control into it. 
```

Reviewer 2: Global comment - apply ` to all API member names.

> *   Added ContextMenu API which allows customization of the webview2 context menu. 

Reviewer 2: ```suggestion
*   Added ContextMenu, which allows customization of the WebView2 context menu. 
```

Reviewer 2: Global comment - WebView2 should be used throughout the doc.

#### Bug fixes  

> *   Improved catching exceptions for host objects in JavaScript.

Reviewer 2: ```suggestion
*   Improved how host objects exceptions are caught in your JavaScript code.
```

> *   Replaced WebView2 icon with generic in Devtools windows.

Reviewer 2: ```suggestion
*   Replaced WebView2 icon with a generic icon in Devtools windows.
```

> *   Re-enable the "Tab" screen sharing option when using `MediaDevices.getDisplayMedia()`. \([\#1566][GithubMicrosoftedgeWebviewfeedbackIssue1566]\)

Reviewer 2: ```suggestion
*   Turn on the Tab screen sharing option when `MediaDevices.getDisplayMedia()` is used. \([\#1566][GithubMicrosoftedgeWebviewfeedbackIssue1566]\)
```

> *   Fixed a bug in the Client Certificate API, where the correct certificate was not getting selected. This is a Runtime change. \([\#1666][GithubMicrosoftedgeWebviewfeedbackIssue1666]\)

Reviewer 2: ```suggestion
*   Fixed a bug in `ClientCertificate` when the correct certificate was not selected. This is a Runtime change. \([\#1666][GithubMicrosoftedgeWebviewfeedbackIssue1666]\)
```

> *   Fixed bug that window.chrome.webview was not available in a new window with the same as parent domain. This change is runtime-specific. \([\#1144][GithubMicrosoftedgeWebviewfeedbackIssue1144]\)

Reviewer 2: ```suggestion
*   Fixed bug where `window.chrome.webview` was unavailable in new windows in the same parent domain. This change is runtime-specific. \([\#1144][GithubMicrosoftedgeWebviewfeedbackIssue1144]\)
```

> *   Fixed a bug where dropdowns would appear behind topmost window. \([\#411][GithubMicrosoftedgeWebviewfeedbackIssue411]\)

Reviewer 2: ```suggestion
*   Fixed a bug where dropdowns menus or lists displayed behind the window with the focus. \([\#411][GithubMicrosoftedgeWebviewfeedbackIssue411]\)
```

> *   Fixed focus issues with put_IsVisible(false). \([\#238][GithubMicrosoftedgeWebviewfeedbackIssue238]\)

Reviewer 2: ```suggestion
*   Fixed focus issues when using `put_IsVisible(false)`. \([\#238][GithubMicrosoftedgeWebviewfeedbackIssue238]\)
```

> *   Fixed bug to make SetVirtualHostNameToFolderMapping apply to popup window.

Reviewer 2: ```suggestion
*   Fixed bug to apply `SetVirtualHostNameToFolderMapping` to popup windows.
```

> *   Fixed bugs where an IDispatch object is returned as IUnknown.

Reviewer 2: ```suggestion
*   Fixed bugs where an `IDispatch` objects were returned as `IUnknown`.
```

#### Promotions

The following APIs are promoted to stable:

> *   IsSwipeNavigationEnabled API.
> *   BrowserProcessExited API
> *   BrowserTaskManager API.  -->

Reviewer 2: ```suggestion
*   `IsSwipeNavigationEnabled`.
*   `BrowserProcessExited`.
*   `BrowserTaskManager`.  -->
```

## 1.0.961.33

Release Date: September 8, 2021  

[NuGet package for WebView2 SDK 1.0.961.33][NuGetGallery1.0.961.33]

For full API compatibility, this version of the WebView2 SDK requires WebView2 Runtime version 93.0.961.33 or higher.

### General  

#### Bug Fixes  

> *   Fixed a bug that was causing ERR_SSL_CLIENT_AUTH_CERT_NEEDED errors. This is a Runtime change.

Reviewer 2: ```suggestion
*   Fixed a bug that caused `ERR_SSL_CLIENT_AUTH_CERT_NEEDED` errors. This is a Runtime change.
```

> *   Fixed a bug that special browser keys like Refresh, Home, Back, etc can't be disabled by AreBrowserAcceleratorKeysEnabled. This change is Runtime-specific.

Reviewer 2: ```suggestion
*   Fixed a bug that special browser keys like **Refresh**, **Home**, **Back**, and so on can't be turned off using `AreBrowserAcceleratorKeysEnabled`. This change is Runtime-specific.
```

> *   Fixed a bug that doesn't render transparent background color.

Reviewer 2: ```suggestion
*   Fixed a bug that doesn't render transparent background colors.
```

Author: n/a

> *   Fixed a bug that caused white flicker when loading WebView2.

Reviewer 2: ```suggestion
*   Fixed a bug that caused the screen to flicker when loading WebView2.
```

Author: n/a

> *   Fixed a bug in WebView2 .NET controls that WebView2 windows are blank when created in the background. \([\#1077][GithubMicrosoftedgeWebviewfeedbackIssue1077]\)

Reviewer 2: ```suggestion
*   Fixed a bug in WebView2 .NET controls where WebView2 windows were blank when created in the background. \([\#1077][GithubMicrosoftedgeWebviewfeedbackIssue1077]\)
```

> *   Fixed a bug where settings were not getting updated for navigations and new window requests to about:blank pages. This is a runtime change.

Reviewer 2: ```suggestion
*   Fixed a bug where settings were not updated when the user navigated to or a new window displayed `about:blank` pages. This is a runtime change.
```

#### Promotions

> The following items are now in stable: 

Reviewer: Change "in stable" to a more standard phrase

[GithubMicrosoftedgeWebviewfeedbackIssue1605]: https://github.com/MicrosoftEdge/WebViewFeedback/issues/1605 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 1605"  
[GithubMicrosoftedgeWebviewfeedbackIssue1678]: https://github.com/MicrosoftEdge/WebViewFeedback/issues/1678 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 1678"  
[GithubMicrosoftedgeWebviewfeedbackIssue1566]: https://github.com/MicrosoftEdge/WebViewFeedback/issues/1566 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 1566"  
[GithubMicrosoftedgeWebviewfeedbackIssue1666]: https://github.com/MicrosoftEdge/WebViewFeedback/issues/1666 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 1666"  
[GithubMicrosoftedgeWebviewfeedbackIssue1144]: https://github.com/MicrosoftEdge/WebViewFeedback/issues/1144 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 1144"  
[GithubMicrosoftedgeWebviewfeedbackIssue411]: https://github.com/MicrosoftEdge/WebViewFeedback/issues/411 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 411"  
[GithubMicrosoftedgeWebviewfeedbackIssue238]: https://github.com/MicrosoftEdge/WebViewFeedback/issues/238 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 238"  

> [GithubMicrosoftedgeWebviewfeedbackIssue1077]: https://github.com/MicrosoftEdge/WebViewFeedback/issues/1077 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 1"  

Reviewer: ```suggestion
[GithubMicrosoftedgeWebviewfeedbackIssue1077]: https://github.com/MicrosoftEdge/WebViewFeedback/issues/1077 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 1077"  
```

[Webview2ReferenceWin32Icorewebview2settings4ViewWebview21090249GetIsgeneralautofillenabled]: /microsoft-edge/webview2/reference/win32/icorewebview2settings4?view=webview2-1.0.902.49&preserve-view=true#get_isgeneralautofillenabled "get_IsGeneralAutofillEnabled - interface ICoreWebView2Settings4 | Microsoft Docs"
[Webview2ReferenceWin32Icorewebview2settings5ViewWebview21090249GetIspinchzoomenabled]: /microsoft-edge/webview2/reference/win32/icorewebview2settings5?view=webview2-1.0.902.49&preserve-view=true#get_ispinchzoomenabled "get_IsPinchZoomEnabled - interface ICoreWebView2Settings5 | Microsoft Docs"
[Webview2ReferenceWin32Icorewebview2experimentalsettings6ViewWebview210955PrereleaseGetHiddenpdftoolbaritems]: /microsoft-edge/webview2/reference/win32/icorewebview2experimentalsettings6?view=webview2-1.0.955-prerelease&preserve-view=true#get_hiddenpdftoolbaritems "get_HiddenPdfToolbarItems - interface ICoreWebView2ExperimentalSettings6 | Microsoft Docs"

> # 1.0.999-prerelease (TBD)

Reviewer: can we put a regular paragraph in between these # and ## headings?   Usually the style guidelines (for regular docs) say to avoid a series of headings without intervening content.

Author: n/a

> ## To experimental

Reviewer: Need clearer heading, or an intro sentence.  What's the nature of items in this section?

Author: Done, this was supposed to be internal comments - went ahead and removed them.

> ## To experimental

Reviewer 2: Is this supposed to be shown in the doc?

[Webview2ReferenceWin32Icorewebview2experimentalframecontentloadingeventhandlerViewWebview210999Prerelease]: /microsoft-edge/webview2/reference/win32/icorewebview2experimentalframecontentloadingeventhandler?view=webview2-1.0.999-prerelease&preserve-view=true "interface ICoreWebView2ExperimentalFrameContentLoadingEventHandler | Microsoft Docs"
[Webview2ReferenceWin32Icorewebview2experimentalframenavigationcompletedeventhandlerViewWebview210999Prerelease]: /microsoft-edge/webview2/reference/win32/icorewebview2experimentalframenavigationcompletedeventhandler?view=webview2-1.0.999-prerelease&preserve-view=true "interface ICoreWebView2ExperimentalFrameNavigationCompletedEventHandler | Microsoft Docs"
[Webview2ReferenceWin32Icorewebview2experimentalframedomcontentloadedeventhandlerViewWebview210999Prerelease]: /microsoft-edge/webview2/reference/win32/icorewebview2experimentalframedomcontentloadedeventhandler?view=webview2-1.0.999-prerelease&preserve-view=true "interface ICoreWebView2ExperimentalFrameDOMContentLoadedEventHandler | Microsoft Docs"

> ## To stable

Reviewer: ```suggestion
<!-- To stable -->
```
Are these intended to be visible headings in the rendered page, or just hidden HTML comment divider lines?

> ## To stable

Reviewer 2: this is also shown in the doc. needed?

[Webview2ReferenceWin32Icorewebview2environment5ViewWebview210999Prerelease]: /microsoft-edge/webview2/reference/win32/icorewebview2environment5?view=webview2-1.0.999-prerelease&preserve-view=true "interface ICoreWebView2Environment5 | Microsoft Docs"
[Webview2ReferenceWin32Icorewebview2environment5ViewWebview210999PrereleaseAddBrowserprocessexited]: /microsoft-edge/webview2/reference/win32/icorewebview2environment5?view=webview2-1.0.999-prerelease&preserve-view=true#add_browserprocessexited "add_BrowserProcessExited - interface ICoreWebView2Environment5 | Microsoft Docs"
[Webview2ReferenceWin32Icorewebview2environment5ViewWebview210999PrereleaseRemoveBrowserprocessexited]: /microsoft-edge/webview2/reference/win32/icorewebview2environment5?view=webview2-1.0.999-prerelease&preserve-view=true#remove_browserprocessexited "remove_BrowserProcessExited - interface ICoreWebView2Environment5 | Microsoft Docs"

> # 1.0.961.33 (TBD)
> 
> ## To stable

Reviewer: ```suggestion
<!-- 1.0.961.33 (TBD) -->

<!-- To stable -->
```

> Run `\\edgefs\users\peiche\release_notes\Get-ApiChanges.ps1 -PreviousReleaseVersion 902m92 -CurrentReleaseVersion 961m93 -AnaheimRepoPath d:\anaheim\src -shouldGenerateLink -BuildAndPatch 961.33` and the links below will be copied to the clipboard.

Reviewer 2: this is shown in the doc. Needed?
